src/infra/src/table/org_invites.rs

```diff
@@ -0,0 +1,400 @@
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use sea_orm::{
+    entity::prelude::*, ColumnTrait, ConnectionTrait, DatabaseBackend, EntityTrait,
+    FromQueryResult, Order, PaginatorTrait, QueryFilter, QueryOrder, QuerySelect, Schema, Set,
+};
+
+use super::get_lock;
+use crate::{
+    db::{connect_to_orm, mysql, postgres, sqlite, IndexStatement, ORM_CLIENT},
+    errors::{self, DbError, Error},
+};
+
+// invitation_id BIGINT PRIMARY KEY,
+// org_id BIGINT NOT NULL,
+// inviter_id BIGINT NOT NULL,
+// invitee_email VARCHAR(255) NOT NULL,
+// role VARCHAR(50) NOT NULL,
+// status ENUM('pending', 'accepted', 'rejected', 'expired') DEFAULT 'pending',
+// token VARCHAR(255) UNIQUE NOT NULL,
+// expires_at TIMESTAMP NOT NULL,
+// created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+
+// define the organizations type
+#[derive(Debug, Clone, Copy, PartialEq, Eq, EnumIter, DeriveActiveEnum)]
+#[sea_orm(rs_type = "i32", db_type = "Integer")]
+pub enum OrgInviteStatus {
+    Pending = 0,
+    Accepted = 1,
+    Rejected = 2,
+    Expired = 3,
+}
+
+// define the org_invites table
+#[derive(Clone, Debug, PartialEq, Eq, DeriveEntityModel)]
+#[sea_orm(table_name = "org_invites")]
+pub struct Model {
+    #[sea_orm(primary_key, auto_increment = true)]
+    pub id: i64,
+    #[sea_orm(column_type = "String(StringLen::N(32))")]
+    pub token: String,
+    #[sea_orm(column_type = "String(StringLen::N(255))")]
+    pub org_id: String,
+    pub inviter_id: String,
+    pub invitee_id: String,
+    #[sea_orm(column_type = "String(StringLen::N(50))")]
+    pub role: String,
+    pub status: OrgInviteStatus,
+    pub expires_at: i64,
+    pub created_at: i64,
+}
+
+#[derive(Copy, Clone, Debug, EnumIter)]
+pub enum Relation {}
+
+impl RelationTrait for Relation {
+    fn def(&self) -> RelationDef {
+        panic!("No relations defined")
+    }
+}
+
+impl ActiveModelBehavior for ActiveModel {}
+
+#[derive(FromQueryResult, Debug)]
+pub struct InvitationRecord {
+    pub org_id: String,
+    pub token: String,
+    pub role: String,
+    pub status: OrgInviteStatus,
+    pub expires_at: i64,
+    pub created_at: i64,
+}
+
+impl InvitationRecord {
+    pub fn new(
+        org_id: &str,
+        token: &str,
+        role: &str,
+        status: OrgInviteStatus,
+        expires_at: i64,
+    ) -> Self {
+        Self {
+            org_id: org_id.to_string(),
+            token: token.to_string(),
+            role: role.to_string(),
+            status,
+            expires_at,
+            created_at: chrono::Utc::now().timestamp_micros(),
+        }
+    }
+}
+
+#[derive(FromQueryResult, Debug)]
+pub struct OrgInviteToken {
+    pub token: String,
+}
+
+pub async fn init() -> Result<(), errors::Error> {
+    create_table().await?;
+    create_table_index().await?;
+    Ok(())
+}
+
```

Comment:
Looks like this new table is only created with enterprise env `O2_CLOUD_ENABLED`, but the CRUD ops & API routes interacting with this table only check for feature flag enterprise. Would enterprise deployment but not `O2_CLOUD_ENABLED` being false causing any issues? 

If this table is needed regardless of `O2_CLOUD_ENABLED`, maybe we should follow the pattern of using sea-orm to generate the `Entity`, just to avoid the potential schema out-of-sync between `Entity` here and the actual table itself.

Reply:
I think, the invitation system should be implemented for cloud mode only. For non-enterprise and enterprise mode, the root users or the admin of orgs should create creds for the users. In case of invitation, the invited user is responsible for auth, the org admin does not create creds for the user. For now, I am removing this part from this pr as this pr focuses on the enterprise version only. Will fix the sea-orm `Entity` creation script in a new pr.